### PR TITLE
MAINT: sparse.linalg: svds should NOT work with lists of lists

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
gh-14299

#### What does this implement/fix?
In gh-14299, I suggested making `svds` work with lists of lists.  I was told to leave it alone. This PR undoes some changes that made `svds` work with lists of lists and reworks the associated tests.

### CODE REVIEW

File: scipy/sparse/linalg/interface.py [hunk 1]
```diff
@@ -823,5 +823,4 @@ def aslinearoperator(A):
                                   rmatmat=rmatmat, dtype=dtype)
 
         else:
-            A = np.atleast_2d(np.asarray(A))
-            return MatrixLinearOperator(A)
+            raise TypeError('type not understood')
```
Comment (mdhaber) [inline scipy/sparse/linalg/interface.py:826]: I reset this file to master. There should not be any changes to this file in #9 after this is merged.

**Comment by mckib2** [comment]
LGTM, feel free to merge when you're ready

File: scipy/sparse/linalg/eigen/tests/test_svds.py [hunk 2]
```diff
@@ -95,21 +95,27 @@ class SVDSCommonTests:
 
     # some of these IV tests could run only once, say with solver=None
 
-    def test_svds_input_validation_A_1(self):
-        message = "invalid shape"
-        with pytest.raises(ValueError, match=message):
-            svds([[[1., 2.], [3., 4.]]], k=1, solver=self.solver)
-
+    def test_svds_input_validation_A(self):
```
Comment (mdhaber) [inline scipy/sparse/linalg/eigen/tests/test_svds.py]: I can parametrize this if you prefer.

**Comment by mdhaber** [comment]
This is your repo : ) I can't merge it. Go for it if you're happy with it, or I can push the `@parametrize`d version if you prefer.

**Comment by mckib2** [comment]
> This is your repo : ) I can't merge it. Go for it if you're happy with it, or I can push the `@parametrize`d version if you prefer.

Whoops...  Forgot you couldn't merge!  Yeah, let's go ahead and do the parametrized version.  Once that's done I'll merge.  I'm working on linting activities now

**Comment by mdhaber** [comment]
Done.

**Pull request merged**